src/Aspire.Hosting/ApplicationModel/ReferenceExpression.cs

```diff
@@ -69,6 +71,15 @@ private ReferenceExpression(string format, IValueProvider[] valueProviders, stri
         return string.Format(CultureInfo.InvariantCulture, Format, args);
     }
 
+    internal static string ParseFormat(string format)
+    {
+        // Escape curly braces which aren't used for a parameter.
+        var parsedFormat = Regex.Replace(format, @"{(?!\d)", "{{");
+        parsedFormat = Regex.Replace(parsedFormat, @"(?<!\d)}", "}}");
```

Comment:
Do we need to use a regex?

Reply:
This expression doesn't have loops and writing the iterative code would probably not be much different, but can we at least use the regex source generator for this?

Reply:
Created GeneratedRegex and moved to new `ReferenceExpressionParser` partial class.

Reply:
Removed RegEx and replaced with logic similar to StringBuilder.AppendFormat.